pyuvdata/__init__.py

```diff
@@ -6,18 +6,55 @@
 
 """
 from __future__ import absolute_import, division, print_function
+import warnings
+from pkg_resources import get_distribution, DistributionNotFound
+
+# Set the version automatically from the package details.
+try:
+    __version__ = get_distribution(__name__).version
```

Comment:
I don't understand how this works, I looked up `get_distribution` but it wasn't at all clear to me that it would get the latest version tag...

Reply:
Ok. But I don't see setup tools_scm or pygitversion listed in any of the new setup.py requirements locations. It seems like those are required for install now?

Reply:
They are both defined in the `pyproject.toml` file. Which i believe is _only_ evaluated during a `pip install ....` call

Reply:
the `setup.py` also has a `use_scm_version: True` or something similar in the setup arguments.

Reply:
ok. I don't know how pip interacts with pyproject.toml. Does putting it there guarantee that it those packages will be listed as required dependences? One thing I just realized: we won't be able to make a conda release of pyuvdata after this change until we make a conda package out of pygitversion.

Reply:
It will make sure that those pacakges exist in the build environment before beginning to build our package. It's like the `install_requires` field in setup.py. We should probably list `pygitversion` as a dependency now though.

Reply:
Ah yeah. Right, so conda packages can only rely on conda packages? Kind of makes sense...

Yes, @plaplant is right about the format.

Reply:
And yes, `pyproject.toml` is _not_ invoked with a `python setup.py install` (clearly, because that is literally running the python interpreter on the `setup.py` module...)

Reply:
Should we leave a `setup_requires` section in the setup.py for people who _don't_ use `pip install`?

Reply:
I really think we should "force" people to use `pip install`. Makes our life (and theirs) easier. Also I don't think `setup_requires` is really supported.

Reply:
We officially _don't_ support installation using `python setup.py install` and require `pip install .`. We broke that functionality in #626 (I believe so that we could support extracting version info without the package needing all dependencies installed first). We also make note of it in the README: https://github.com/RadioAstronomySoftwareGroup/pyuvdata#optionally-install-the-development-version. Perhaps now that we're using `pygitversion` we can make it so that we support `python setup.py install` again, though clearly our users haven't been clamoring to use that particular installation option, so I'm inclined to leave things as they are.

Reply:
I had forgotten about that. If we already don't support it and there is no big outcry, I'm not really bothered by `pygitversion` only appearing in the `pyproject.toml` file. `pip` will find it during the PEP517 processesing and build with pygitversion.